src/appleseed/foundation/array/meshalgorithm.cpp

```diff
@@ -0,0 +1,148 @@
+
```

Comment:
I don't think `foundation/array/` is the right place for mesh algorithms. Once #2347 is done, we should move these files to `foundation/geometry/`.